client/src/modules/cohort/index.jsx

```diff
@@ -98,21 +124,30 @@ class Cohort extends PureComponent {
 
   hideDialog = () => this.handleDialogContext(null)();
 
-  handleArchivedListsVisibility = id => () => {
-    const { areArchivedListVisible } = this.state;
-    this.setState({ areArchivedListVisible: !areArchivedListVisible });
-    this.handleArchivedListsData(id);
-  };
+  handleArchivedListsVisibility = id => () =>
+    this.setState(
+      ({ areArchivedListsVisible }) => ({
+        areArchivedListsVisible: !areArchivedListsVisible
+      }),
+      () => this.handleArchivedListsData()
+    );
 
   handleArchivedListsData = id => {
-    const { areArchivedListVisible } = this.state;
+    const { areArchivedListsVisible } = this.state;
     const {
       fetchArchivedListsMetaData,
       removeArchivedListsMetaData
     } = this.props;
-    !areArchivedListVisible
-      ? fetchArchivedListsMetaData(id)
-      : removeArchivedListsMetaData();
+
+    if (areArchivedListsVisible) {
+      this.setState({ pendingForArchivedLists: true }, () => {
+        fetchArchivedListsMetaData()
+          .then(() => this.setState({ pendingForArchivedLists: false }))
+          .catch(() => this.setState({ pendingForArchivedLists: false }));
+      });
+    } else {
```

Comment:
check if this else is necessary

Reply:
In my opinion, in this case, it's necessary. This method has to decide what to do depending on areArchivedCohortsVisible variable, but in one case we have to do more than one thing so I had to use if else statement.